extensions/wikia/Search/classes/Config.php

```diff
@@ -873,82 +1006,88 @@ public function getTruncatedResultsNum( $formatted = false ) {
 		if ( $formatted ) {
 			$result = $this->getService()->formatNumber( $result );
 		}
+
 		return $result;
 	}
 
 	/**
 	 * These search profiles are used to figure out what tab we're on and how we should be searching based on that
 	 * While kind of a view concern, moved here so it can play nicely with the namespaces value
+	 *
 	 * @return array
 	 */
 	public function getSearchProfiles() {
 		$nsAllSet = array_keys( $this->getService()->getSearchableNamespacesFromSearchEngine() );
 		$defaultNamespaces = $this->getService()->getDefaultNamespacesFromSearchEngine();
 
-	    $profiles = array(
-	            SEARCH_PROFILE_DEFAULT => array(
-	                    'message' => 'wikiasearch2-tabs-articles',
-	                    'tooltip' => 'searchprofile-articles-tooltip',
-	                    'namespaces' => $defaultNamespaces,
-	                    'namespace-messages' => $this->getService()->getTextForNamespaces( $defaultNamespaces ),
-	            ),
-	            SEARCH_PROFILE_IMAGES => array(
-	                    'message' => 'wikiasearch2-tabs-photos-and-videos',
-	                    'tooltip' => 'searchprofile-images-tooltip',
-	                    'namespaces' => array( NS_FILE ),
-	            ),
-	            SEARCH_PROFILE_USERS => array(
-	                    'message' => 'wikiasearch2-users',
-	                    'tooltip' => 'wikiasearch2-users-tooltip',
-	                    'namespaces' => array( NS_USER )
-	            ),
-	            SEARCH_PROFILE_ALL => array(
-	                    'message' => 'searchprofile-everything',
-	                    'tooltip' => 'searchprofile-everything-tooltip',
-	                    'namespaces' => $nsAllSet,
-	            ),
-	            SEARCH_PROFILE_ADVANCED => array(
-	                    'message' => 'searchprofile-advanced',
-	                    'tooltip' => 'searchprofile-advanced-tooltip',
-	                    'namespaces' => $this->getNamespaces(),
-	                    'parameters' => array( 'advanced' => 1 ),
-	            )
-	    );
-
-	    $this->getService()->invokeHook( 'SpecialSearchProfiles', array( &$profiles ) );
-
-	    foreach ( $profiles as $key => &$data ) {
-	        sort( $data['namespaces'] );
-	    }
-
-	    return $profiles;
+		$profiles = [
+			SEARCH_PROFILE_DEFAULT => [
+				'message' => 'wikiasearch2-tabs-articles',
+				'tooltip' => 'searchprofile-articles-tooltip',
+				'namespaces' => $defaultNamespaces,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "namespaces" 8 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)